archive/100-theorems-list/93_birthday_problem.lean

```diff
@@ -0,0 +1,22 @@
+/-
+Copyright (c) 2021 Eric Rodriguez. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Eric Rodriguez
+-/
+import data.fintype.card_embedding
+
+/-!
+# Birthday Problem
+
+This file proves Theorem 93 from the [100 Theorems List](https://www.cs.ru.nl/~freek/100/).
+
+As of writing, probability is not implemented in Lean, so we instead state the birthday problem
+in terms of injective functions. The general result about `fintype.card (α ↪ β)` which this proof
+uses is `fintype.card_embedding`.
```

Comment:
I am speaking from ignorance about this part of the library here, but we have counting measure, and of course we have measures of sets, so it is perfectly possible to state things like "the measure of the injective functions is < 1/2". It just seems an overly strong statement "probability in not implemented in Lean"!

Reply:
I wasn't actually aware of this; I read somewhere that "lean was not ready for probability" but clearly was misguided!

Reply:
Certainly it has plenty of *measure theory*. Perhaps someone was trying to make the distinction between measure theory per se and how probabilists think in terms of random variables, with "no underlying measure space".